testing/typescript/data/custom_store.ts

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+// eslint-disable-next-line import/no-extraneous-dependencies
+import $ from 'jquery';
+import { HttpClient } from '@angular/common/http';
+import CustomStore from '../../../js/data/custom_store';
+
+export function loadByFetch(url: string): CustomStore {
+  return new CustomStore({
+    load: () => fetch(url).then((response) => response.json()),
+  });
+}
+
+export function loadByFetchWithModification(url: string): CustomStore {
+  return new CustomStore({
+    load: () => fetch(url)
+      .then((response) => response.json())
+      .then((data) => ({
+        data: data.data,
+        totalCount: data.totalCount,
+        summary: data.summary,
+        groupCount: data.groupCount,
+      })),
+  });
+}
```

Comment:
This test is also seems to be useless, since the `load` accepts `() => Promise<object>` as a value.
However, we can test that type inference works fine when the `load` fields contains a lambda that returns one of the known types:
```suggestion
export async function infersTItemFromComplexLoadResult() {
  const store = new CustomStore({
    load: async () => ({
      data: ['1', '2', '3'],
      totalCount: 10,
      summary: [10],
      groupCount: 3,
    })
  });
  let value: string;
  //if TItem inferred incorrectly, the assignment below will result to compilation error.
  //However, if TItem inferred as any, it still will be broken, so we may try adding some logic to avoid it
  value = await store.byKey('stub');
}
```

Reply:
one-liner?
```ts
const value: string = await store.byKey('stub');
```

Reply:
Maybe it's better to use more complex type instead of built-in:

```
      data: [ { a: 123 } ],
```
```
  const value: { a: number } = await store.byKey('stub');
```